js/src/dapps/registry/register/actions.js

```diff
@@ -0,0 +1,35 @@
+const sha3 = window.parity.api.format.sha3;
+const toWei = window.parity.api.format.toWei;
+
+export const start = (name) => ({ type: 'register start', name });
+
+export const success = (name) => ({ type: 'register success', name });
+
+export const fail = (name) => ({ type: 'register error', name });
+
+export const register = (name) => (dispatch, getState) => {
+  const { contract, account } = getState();
+  if (!contract || !account) return;
+  const reserve = contract.functions
+    .find((f) => f.name === 'reserve');
+
```

Comment:
could also so (shorter, especially if you do a lot)

`const reserve = contract.instance.reserve`